spring-integration-test/src/main/java/org/springframework/integration/test/mock/MockMessageHandler.java

```diff
@@ -26,19 +24,39 @@
 import org.hamcrest.Matcher;
 
 import org.springframework.integration.handler.AbstractMessageProducingHandler;
+import org.springframework.integration.test.matcher.PayloadAndHeaderMatcher;
+import org.springframework.integration.test.matcher.PayloadMatcher;
 import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
 
 /**
+ * The {@link AbstractMessageProducingHandler} extension for the assertion purpose in tests.
+ * <p>
+ * The provided {@link Matcher}s are applied to the incoming messages one at a time
+ * until the last {@link Matcher}, which is applied for all subsequent messages -
+ * the similar behavior exists in the
+ * {@code Mockito.doReturn(Object toBeReturned, Object... toBeReturnedNext)}.
+ * <p>
+ * Typically is used as a chain of assertions and optional replies for them:
+ * <pre class="code">
+ * {@code
+ *      MockIntegration.mockMessageHandler(hasHeader("bar", "BAR"))
+ *               .thenReply()
+ *               .assertNext(new GenericMessage<>("foo", Collections.singletonMap("key", "value")))
+ *               .assertNext(hasPayload("foo"))
```

Comment:
I think *then* has a comprehensive meaning if we have a good outcome from the assertion.

I don't know what better can we have here, but I feel comfortable with the current API...